### DISCUSSION TITLE
HM-CC-TC Change Operation Mode

### DISCUSSION THREAD

**Discussion by shortyxs** [SukramJ/aiohomematic#1677] [Q&A]
Hi everybody and many thanks for providing this awsome extension to Home Assistant.

I have many of the old HM-CC-TC Thermostats and these as far as I know do not allow to change the mode via XMLRPC. However with some other Homeautomation software they somehow managed to change that mode from Auto to Manual or Central. I assueme back in the day changeing the mode was more seen like a Setup thing rather than beeing something you want to use.
At least I did not find anything in the entities, that looks something like an operation/ heating mode.
The slightly younger HM-CC-RT-DN has all that functions.

I don't even remember exactly if that was ioBorker or FHEM.
Do you see any chance of implementing this feature into the homematic HACs?

Here is something I found on the [homeatic forum](https://homematic-forum.de/forum/viewtopic.php?f=31&t=5621)
In this TCL script it looks like it is just calling localhost, that same thing should be possible from a remote host I guess.
However as a Sys-Admin I know how to use ans lightly change stuff but I'm unable to do real software development.

Or am I limited by my also very old CCU2, still in use? 

My goal is to set a manual mode with a low target temperature, if the weather forecast next 8h will have a higher temeprature outside temperature. To optimize the summer season.

Any help is much appreciated.

**Comment by Baxxy13** [comment]
Sieht jetzt erstmal unspektakulär aus. Aber gut 2010 ist auch lange her und die Welt dreht sich weiter. 
Wurden die HM-CC-TC erfolgreich (korrekt) integriert?

Wenn ich das richtig im Forum gelesen habe muss der MASTER-PARAMETER  `MODE_TEMPERATUR_REGULATOR` in Kanal:2 des Gerätes auf 0/1/2 für die 3 Modi gesetzt werden. 

In HA -> Entwicklerwerkzeuge -> Aktionen -> YAML-Modus...
```yaml
action: homematicip_local.put_paramset
data:
  device_id: 17e2ec4b65c40e44152609aaa4ef6ac6
  paramset_key: MASTER
  paramset:
    MODE_TEMPERATUR_REGULATOR: 1
  channel: 2
```
... einfügen, zurück zum UI-Modus wechseln und dein Gerät auswählen, dann "Aktion ausführen".

Sollte grundsätzlich klappen.
